packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -118,6 +118,11 @@ class ImageCache {
       _currentSizeBytes -= image.sizeBytes;
       return true;
     }
+    final _PendingImage pending = _pendingImages.remove(key);
```

Comment:
Is that desired that an `evict` also aborts an in-progress download? If so, we should probably update the doc comment to mention that?

Reply:
This shouldn't abort an in-progress download. If a new completer is created, there are two listeners attached to it: one from the cache to store the Completer w/ size information, and one from the widget that originally requested the completer.

Removing this from pending and removing the listener only prevents the cache completer from completing, it should load into the widget normally.

Reply:
Ah! That makes sense. Sorry for the misunderstanding.